src/Mvc/Mvc.Core/src/JsonResult.cs

```diff
@@ -12,7 +14,7 @@ namespace Microsoft.AspNetCore.Mvc
     /// <summary>
     /// An action result which formats the given object as JSON.
     /// </summary>
-    public class JsonResult : ActionResult, IStatusCodeActionResult
+    public class JsonResult : ActionResult, IResult, IStatusCodeActionResult
```

Comment:
It will be interesting to see which of MVC's result types can and can't get represented as `IResult`. `ObjectResult` comes to mind.